### ISSUE TITLE
Allow Fido searches without time

### ISSUE THREAD

**Issue by dstansby** [sunpy/sunpy#3830]
I am currently trying to download some synoptic maps from JSOC. Because they is one map / Carrington rotation, they are indexed by Carrington number and *not* time. This means that the following search should be valid, but instead throws an error:

```python
from sunpy.net import Fido, attrs as a
import drms
c = drms.Client()  
print(c.pkeys('hmi.synoptic_mr_polfil_720s'))
res = Fido.search(a.jsoc.Series('hmi.synoptic_mr_polfil_720s'),
                  a.jsoc.PrimeKey('CAR_ROT', '2215'),
                  a.jsoc.Notify('sunpy@sunpy.org'))
```

The print statement prints `['CAR_ROT']`, I think confirming that time is not needed.

The error thrown is
```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-38-5774b7b06731> in <module>
      5 res = Fido.search(a.jsoc.Series('hmi.synoptic_mr_polfil_720s'),
      6                   a.jsoc.PrimeKey('CAR_ROT', '2215'),
----> 7                   a.jsoc.Notify('sunpy@sunpy.org'))  

~/github/sunpy/sunpy/net/fido_factory.py in search(self, *query)
    318         """  # noqa
    319         query = attr.and_(*query)
--> 320         return UnifiedResponse(query_walker.create(query, self))
    321 
    322     def fetch(self, *query_results, path=None, max_conn=5, progress=True,

~/github/sunpy/sunpy/net/attr.py in create(self, *args, **kwargs)
    457         Call the create function(s) matching the arguments to this method.
    458         """
--> 459         return self.createmm(self, *args, **kwargs)
    460 
    461     def apply(self, *args, **kwargs):

~/github/sunpy/sunpy/util/functools.py in wrapper(*args, **kwargs)
     17 
     18     def wrapper(*args, **kwargs):
---> 19         return dispatcher.dispatch(args[1].__class__)(*args, **kwargs)
     20 
     21     wrapper.dispatch = dispatcher.dispatch

~/github/sunpy/sunpy/net/fido_factory.py in _create_and(walker, query, factory)
    249         for at in query.attrs:
    250             error += str(at) + ', '
--> 251         raise ValueError(error)
    252 
    253     # Return the response and the client

ValueError: The following part of the query did not have a time specified:
<Series('hmi.synoptic_mr_polfil_720s')>, <PrimeKey('CAR_ROT','2215')>, <Notify('sunpy@sunpy.org')>, 
```

**Comment by dstansby** [comment]
Commenting out the lines that check for the precense of time:
https://github.com/sunpy/sunpy/blob/ffd9c0c01a91915fb793ca3104676f806d21843b/sunpy/net/fido_factory.py#L246-L251

fixes this issue, and the query is completed as expected.

**Comment by Cadair** [comment]
This should be addressed by #3770 

There are actually some interesting fallout issues from changing this in fido_factory. VSO and various data retriever sources need to only respond to queries with time in them.
